soh/soh/Enhancements/randomizer/randomizer_check_tracker.cpp

```diff
@@ -0,0 +1,1046 @@
+        ret = true;
+
+    slowCheckIdx++;
+    return ret;
+}
+
+bool ShouldUpdateChecks() {
+    // TODO eventually will need to be hooked into game elements rather than just save file
+    if (CVar_GetS32("gCheckTrackerOptionPerformanceMode", 0))
+        return SlowUpdateCheck();
+    else
+        return true;
+}
+
+void UpdateInventoryChecks() {
+    //For all the areas with compasses, if you have one, spoil the area
+    for (u8 i = RCAREA_DEKU_TREE; i <= RCAREA_ICE_CAVERN; i++)
+        if (gSaveContext.inventory.dungeonItems[i - RCAREA_DEKU_TREE] & 0x02)
+            areasSpoiled |= (1 << i);
+}
+
+void UpdateChecks() {
+    int idx = 0;
+    RandomizerCheckObject* lastCheck;
+    RandomizerCheckShow lastStatus;
+    for (auto& rcObj : checks) {
+        RandomizerCheckShow* checkStatusPtr = &checkStatusMap.find(rcObj.rc)->second;
+        lastStatus = *checkStatusPtr;
+        *checkStatusPtr = GetCheckStatus(rcObj, idx);
+
+        //Update areasFullyChecked 
+        if (lastStatus != *checkStatusPtr) {
+            if (lastStatus != RCSHOW_CHECKED && lastStatus != RCSHOW_SAVED && (*checkStatusPtr == RCSHOW_CHECKED || *checkStatusPtr == RCSHOW_SAVED))
+                areaChecksGotten[rcObj.rcArea]++;
+            else if ((lastStatus == RCSHOW_CHECKED || lastStatus == RCSHOW_SAVED) && *checkStatusPtr != RCSHOW_CHECKED && *checkStatusPtr != RCSHOW_SAVED)
+                areaChecksGotten[rcObj.rcArea]--;
+
+            if (areaChecksGotten[rcObj.rcArea] == areaChecksTotal[rcObj.rcArea])
+                areasFullyChecked |= (1 << rcObj.rcArea);
+            else
+                areasFullyChecked &= (0xFFFFFFFF - (1 << rcObj.rcArea));
+
+            if (areaChecksGotten[rcObj.rcArea] != 0 || RandomizerCheckObjects::AreaIsOverworld(rcObj.rcArea))
+                areasSpoiled |= (1 << rcObj.rcArea);
+        }
+
+        lastCheck = &rcObj;
+
+        idx++;
+    }
+}
+
+void UpdateOrdering(bool init) {
+    // Sort the entire thing
+    if (init) {
+        std::sort(checks.begin(), checks.end(), CompareCheckObject);
+        return;
+    }
+    
+    //sort each area individually
+    int startOffset = 0;
+    int endOffset = 0;
+    for (int x = 0; x < sizeof(areaChecksTotal); x++) {
+        endOffset = startOffset + areaChecksTotal[x];
+        std::sort(checks.begin() + startOffset, checks.begin() + endOffset, CompareCheckObject);
+        startOffset += areaChecksTotal[x];
+    }
+}
+
+bool CompareCheckObject(RandomizerCheckObject i, RandomizerCheckObject j) {
+    if (i.rcArea < j.rcArea)
+        return true;
+    else if (i.rcArea > j.rcArea)
+        return false;
+
+    if (checkStatusMap.find(i.rc)->second < checkStatusMap.find(j.rc)->second)
+        return true;
+    else if (checkStatusMap.find(i.rc)->second > checkStatusMap.find(j.rc)->second)
+        return false;
+
+    if (i.rcShortName < j.rcShortName)
+        return true;
+    if (i.rcShortName > j.rcShortName)
+        return false;
```

Comment:
is there a way to avoid string comparisons here?

Reply:
The goal of the string comparisons is to show randomizer checks in a deterministic sort order. I can replace the sorting with the RandomizerCheck, which will still give determinism.

Reply:
Resolved by 12749935f6260b5e6cb057826e9caa12653d5689